frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,384 @@
+
+		if (!nonProtectedBranches.includes(input)) {
+			addToast({
+				message: `Changed branch name to "${input}".`,
+				type: ToastType.Success,
+				dismissible: true,
+				timeout: 1500
+			});
+			return;
+		}
+
+		// Call backend to update working directory by checking out the branch
+		const response = await fetch(`${apiAddress}/api/check-out/branches/${input}`, {
+			method: 'PUT',
+			credentials: 'include'
+		});
+
+		if (!response.ok) {
+			addToast({
+				message: `Failed to check out branch. Error ${response.status}: ${response.statusText}`,
+				type: ToastType.Error,
+				dismissible: true
+			});
+			return;
+		}
+
+		// After checking out, call the pull endpoint
+		const pullResponse = await fetch(`${apiAddress}/api/pull/${input}`, {
+			method: 'POST',
+			credentials: 'include'
+		});
+
+		if (!pullResponse.ok) {
+			addToast({
+				message: `Failed to pull latest changes for branch "${input}".`,
+				type: ToastType.Error,
+				dismissible: true
+			});
+			return;
+		} else {
+			addToast({
+				message: `Branch "${input}" checked out and updated successfully.`,
+				type: ToastType.Success,
+				dismissible: true,
+				timeout: 1200
+			});
+		}
+
+		// Fetch the updated document tree after pulling changes
+		const treeResponse = await fetch(`${apiAddress}/api/tree`, {
+			method: 'GET',
+			credentials: 'include'
+		});
+
+		if (treeResponse.ok) {
+			const updatedTree = await treeResponse.json();
+			documentTreeStore.set(updatedTree); // Update the store with the new tree
+
+			cache.flush();
+
+			// After updating the tree, check if there's a current file
+			const currentFilePath = get(currentFile);
+			if (currentFilePath) {
+				// Fetch the content of the current file
+				const fileContentResponse = await fetch(
+					`${apiAddress}/api/doc?path=${encodeURIComponent(currentFilePath)}`,
+					{
+						method: 'GET',
+						credentials: 'include'
+					}
+				);
+
+				if (fileContentResponse.ok) {
+					const fileContent = await fileContentResponse.json(); // Get the content of the file
+					editorText.set(fileContent.contents); // Update the editor text
+					cache.set(currentFilePath, fileContent.contents);
+				} else {
+					console.error(
+						'Failed to fetch the file content:',
+						fileContentResponse.status,
+						fileContentResponse.statusText
+					);
+				}
+			}
+		} else {
+			console.error(
+				'Failed to fetch updated document tree:',
+				treeResponse.status,
+				treeResponse.statusText
+			);
+		}
+	}
+
+	function toggleMenu() {
+		showMenu = !showMenu;
+	}
+
+	function closeMenu() {
+		showMenu = false;
+		showInput = false;
+	}
+</script>
+
+<div class="branch-dropdown">
+	<button on:click={toggleMenu} class="branch-button" title="Set Branch Name">
+		{$currentBranch.length > 100 ? `${$currentBranch.slice(0, 100)}...` : $currentBranch}
+	</button>
+
+	{#if showMenu}
+		<div class="branch-menu">
+			<button class="close-button" on:click={closeMenu} aria-label="Close menu">✖</button>
```

Comment:
```suggestion
			<button class="close-button" on:click={closeMenu} aria-label="Close menu">
			  	<svg
				xmlns="http://www.w3.org/2000/svg"
				height="1.5rem"
				viewBox="0 -960 960 960"
				width="1.5rem"
				role="none"
			>
				<title>Exit</title>
				<path
					d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z"
				/></svg
			>
			</button>
```
I would prefer if this were an svg, you'll probably need to do some css tweaks. (better formatted though, just run `npm run format`